public/app/features/dashboard-scene/scene/PlaylistBehavior.test.ts

```diff
@@ -0,0 +1,77 @@
+import { SceneGridItem, SceneGridLayout, VizPanel } from '@grafana/scenes';
+import { playlistSrv } from 'app/features/playlist/PlaylistSrv';
+
+import { DashboardScene } from './DashboardScene';
+import { PlaylistBehavior } from './PlaylistBehavior';
+
+jest.mock('app/features/playlist/PlaylistSrv', () => ({
+  playlistSrv: {
+    isPlaying: true,
+    start: jest.fn(),
+    next: jest.fn(),
+    prev: jest.fn(),
+    stop: jest.fn(),
+  },
+}));
+
+describe('PlaylistBehavior', () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
+    playlistSrv.isPlaying = true;
+  });
+
+  it.each([true, false])('should set isPlaying to the playingSrv.isPlaying when activated', async (isPlaying) => {
+    playlistSrv.isPlaying = isPlaying;
+    const { behavior } = await buildTestScene();
+
+    expect(behavior.state.isPlaying).toBe(isPlaying);
+  });
+
+  it('should call playlistSrv.next() when next() is called', async () => {
+    const { behavior } = await buildTestScene();
+    behavior.next();
+    expect(playlistSrv.next).toHaveBeenCalled();
+  });
+
+  it('should call playlistSrv.prev() when prev() is called', async () => {
+    const { behavior } = await buildTestScene();
+    behavior.prev();
+    expect(playlistSrv.prev).toHaveBeenCalled();
+  });
+
+  it('should call playlistSrv.stop() when stop() is called', async () => {
+    const { behavior } = await buildTestScene();
+    behavior.stop();
+    expect(playlistSrv.stop).toHaveBeenCalled();
+    expect(behavior.state.isPlaying).toBe(false);
+  });
+});
+
+async function buildTestScene() {
+  const behavior = new PlaylistBehavior({});
+  const scene = new DashboardScene({
+    title: 'My dashboard',
+    uid: 'dash-1',
+    $behaviors: [behavior],
```

Comment:
I would put this in normal prop instead of a behavior, simpler and requires less type casting / instance of checks or lookup functions.

The behaviors are mostly for ways to extend function/behavior or library components

Reply:
I agree it needs less casting and getters, on the other side this keeps the playlist behavior out of the scene main class, and everything is managed in a very simple way from a specific "behavior".

No strong opinions here, I implemented it as part of the scene here https://github.com/grafana/grafana/pull/83551

IMHO,  I think this leaves a specific feature (or behavior) out of a main dashboard logic, encapsulated and out of the big class of DashboardScene and easily traceable.